src/pexprs-assertAllApplicationsAreValid.js

```diff
@@ -13,9 +13,11 @@ var pexprs = require('./pexprs');
 // --------------------------------------------------------------------
 
 var lexifyCount;
+var syntactifyCount;
```

Comment:
Now that we have both `#` and `@`, I think what we need to do here is maintain a stack, just like `inLexifiedContextStack` in the State object. `foo = @(#Bar)` should still result in an error -- it is invoking a syntactic rule from a lexical context.